Gifski/EditVideoViewController.swift

```diff
@@ -318,10 +342,43 @@ final class EditVideoViewController: NSViewController {
 			}
 		}()
 		let frameCount = duration * frameRateSlider.doubleValue
-		let dimensions = resizableDimensions.changed(dimensionsType: .pixels).currentDimensions.value
-		var fileSize = (Double(dimensions.width) * Double(dimensions.height) * frameCount) / 3
-		fileSize = fileSize * (qualitySlider.doubleValue + 1.5) / 2.5
-		estimatedSizeLabel.stringValue = "Estimated size: " + formatter.string(fromByteCount: Int64(fileSize))
+
+		estimatedSizeLabel.textColor = NSColor(red: 0.13, green: 0.13, blue: 0.13, alpha: 1.0)
+		estimatedSizeLabel.stringValue = "Starting Estimation..."
+		estimateWorkItem = DispatchWorkItem {
+			print("Total: \(frameCount)")
+
+			Gifski.setOnNewFrame { framesWritten in
+				DispatchQueue.main.sync {
+					self.estimatedSizeLabel.textColor = NSColor(red: 0.11, green: 0.37, blue: 0.13, alpha: 1.0)
+				}
+
+				if self.dataSize == 0 {
+					return
+				}
+
+				self.framesWritten = framesWritten
+				DispatchQueue.main.sync {
+					self.estimatedSizeLabel.stringValue = "Estimated size: " + self.formatter.string(fromByteCount: Int64(self.dataSize * (Int64(frameCount) / Int64(framesWritten))))
+				}
+			}
+			Gifski.setWriteCallback { size in
```

Comment:
also, how about having these two callbacks as arguments for the `run(conversion)` function? I don't like these static properties as they can be replaced any time and this would resolve in an undefined behavior by calling old conversion on new callbacks